src/main/java/bio/terra/service/snapshot/SnapshotService.java

```diff
@@ -276,6 +277,11 @@ void validateForByRequestIdMode(SnapshotRequestContentsModel snapshotRequestCont
       throw new ValidationException(
           "Snapshot Create Flight with id %s is still running".formatted(flightId));
     }
+    var requesterEmail = snapshotAccessRequest.getCreatedBy();
```

Comment:
Is there a reason why we do these validation here instead of in `SnapshotRequestValidator`? That might be a better place to do this kind of validation, since other validations are being done there.

Reply:
I don't think we would have the details form the snapshotAccessRequest in the `SnapshotRequestValidator`.

Reply:
It looks like we currently don't inject services into our validators. They're spring Components so we should be able to do anything we do in the controller though. I don't know if there's a reason to do it one place over the other; right now it's OK to leave it here